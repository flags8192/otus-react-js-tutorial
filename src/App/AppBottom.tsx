import React, { FC, memo, useContext } from 'react';
import Panel from '@src/layout/Panel';
import styled from '@emotion/styled';
import store from '@src/store';

const Message = styled.p`
  margin: auto 20px;
`;

const AppBottom: FC = () => {
  const { state } = useContext(store);
  const { gameFieldPercentFilled } = state;

  return (
    <Panel role="bottomPanel">
      <Message role="messageResult">Результат:</Message>
      <Message role="messagePercentFilled">{`Процент заполнения: ${gameFieldPercentFilled}`}</Message>
    </Panel>
  );
};

export default memo(AppBottom);
