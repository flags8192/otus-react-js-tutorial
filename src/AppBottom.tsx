import React, { PureComponent } from 'react';
import styled from '@emotion/styled';

const Root = styled.div`
  height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

interface IAppBottom {
  countUsers: number;
}
class AppBottom extends PureComponent<IAppBottom> {
  render() {
    return (
      <Root>
        <p role="userCount">Количество пользователей: {this.props.countUsers}</p>
      </Root>
    );
  }
}

export default AppBottom;
