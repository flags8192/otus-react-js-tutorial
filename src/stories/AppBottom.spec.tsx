import React from 'react';
import { render, screen } from '@testing-library/react';
import AppBottom from '@src/AppBottom';

describe('AppBody', () => {
  it('Render <AppBottom> without data', () => {
    const { asFragment } = render(<AppBottom countUsers={50} />);
    expect(asFragment()).toMatchSnapshot();
    const dataMessage = screen.getByText(/Количество пользователей: 50/gi);
    expect(dataMessage).toBeInTheDocument();
  });
});
