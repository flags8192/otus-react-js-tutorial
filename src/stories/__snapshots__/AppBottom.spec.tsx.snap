// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody Render <AppBottom> without data 1`] = `
<DocumentFragment>
  <div
    class="css-1co368a"
  >
    <p
      role="userCount"
    >
      Количество пользователей: 50
    </p>
  </div>
</DocumentFragment>
`;
